src/EnergyPlus/DaylightingManager.cc

```diff
@@ -5816,7 +5816,7 @@ void DayltgHitObstruction(EnergyPlusData &state,
     } else { // Surface octree search
 
         auto const &window_base(window_iBaseSurf > 0 ? state.dataSurface->Surface(window_iBaseSurf) : window);
-        auto const window_base_p(&window_base);
+        auto const *window_base_p(&window_base);
```

Comment:
The compiler was smart enough to do this automatically? This looks like another opportunity for a custom check but honestly this is the first time I have seen this.

Reply:
Yes, it was. Because pointers have different type signatures than references. I would be okay with referring to pointers as just `auto const`, except that we make so much use of references that it would make it impossible to tell between a legitimate pointer variable or a reference that should really be `auto const &`. Have I mentioned yet how `auto` and `&` are terrible together?